awx/main/tasks.py

```diff
@@ -1720,8 +1736,13 @@ def final_run_hook(self, job, status, private_data_dir, fact_modification_times,
                 os.path.join(private_data_dir, 'artifacts', str(job.id), 'fact_cache'),
                 fact_modification_times,
             )
-        if isolated_manager_instance:
+        if isolated_manager_instance and not job.is_containerized:
             isolated_manager_instance.cleanup()
+
+        if job.is_containerized:
+            from awx.main.scheduler.kubernetes import PodManager # prevent circular import
+            PodManager(job).delete()
```

Comment:
We should add some exception handling around this.

```python
try:
    ...
except Exception:
    logger.exception('Failed to clean up pod.')
```

Reply:
Not ignoring this, but will probably leave for a more comprehensive PR focused on logging. Definitely on my list.